### PR TITLE
generic proxy: add new request matcher to simplify route table

### PR DESCRIPTION
**Pull request by wbpcode**
Commit Message: generic proxy: add new request matcher to simplify route table
Additional Description:

A new custom matcher for generic proxy is added to simplify the route table. When simple AND semantic is used, the users needn't write complex configuration to combine different input/match.

Risk Level: low.
Testing: unit.
Docs Changes: n/a.
Release Notes: n/a.
Platform Specific Features: n/a.

### CODE REVIEW

**Review by htuch — COMMENTED** [review]
@soulxu can you review as code owner? Happy to merge when you are good here, no API impact.

**Review by soulxu — APPROVED** [review]
This LGTM, thanks

**Comment by wbpcode** [comment]
friendly ping @htuch

**Review by htuch — APPROVED** [review]

**Pull request merged**
